### PR TITLE
fix gen-owl: move id, name, and type to prov.yaml

### PR DESCRIPTION
**Pull request by dwinston**
To close biolink/biolinkml#333.

gen-owl schema/{prov,core,nmdc}.yaml all work after this change (using `biolinkml==1.7.0`).

### CODE REVIEW

**Comment by cmungall** [comment]
This will take a bit of discussion. I dont think that are properties like id and name belong in prov.

**Comment by dwinston** [comment]
Sure. You recently [added id and name to prov](https://github.com/microbiomedata/nmdc-metadata/commit/c760c6ff4c5e5d676ce0f4234238298e1ade3c6d), but I guess this was intended as temporary? I'm fine with id and name not being in prov -- it makes sense for prov to merely subset terms of <http://www.w3.org/ns/prov#> and not add anything else.

**Comment by cmungall** [comment]
so what I did was add these as class slots to the classes in prov, the
fields are still defined in core

this is suboptimal for a modularization point of view as we have a
reciprical dependency between prov and core. this is harmless so long as
you are using the root of the import chain (nmdc), which is what we care
about... but if you want to work with prov as a separate unit then you
don't have the fields declared and you get errors as you discovered

tl;dr - not a big issue for now but we can fix later with better
modulatization!

**Comment by dwinston** [comment]
Ah, now I understand the adding/definition distinction and how that relates to the dependency resolution. Agreed that nmdc.yml is the principal concern. Thanks for clarifying.

**Comment by dwinston** [comment]
not a priority. can revisit.
